From b8f8e626877bc32412c18294a72c4e3c60f733bc Mon Sep 17 00:00:00 2001
From: Philippe Nunes <philippe.nunes@linux.intel.com>
Date: Thu, 22 Nov 2012 17:34:28 +0100
Subject: [PATCH 19/23] s_sat.c: Fix envelope cmd and enable Setup Event List
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

The format specifier used to build the envelope command is now x86 compatible
If the Setup Event List proactive command is not supported, it prevents to
initiate a SAT session. So, I enabled the proactive command even though
events are not monitored.
---
 src/s_sat.c |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/s_sat.c b/src/s_sat.c
index 774388c..4a18518 100644
--- a/src/s_sat.c
+++ b/src/s_sat.c
@@ -221,7 +221,7 @@ static gboolean on_event_sat_proactive_command(CoreObject *o, const void *event_
 		dbg("wrong input");
 		break;
 	}
-	if ((decoded_data.cmd_type == SAT_PROATV_CMD_REFRESH) || (decoded_data.cmd_type == SAT_PROATV_CMD_SETUP_EVENT_LIST)) {
+	if (decoded_data.cmd_type == SAT_PROATV_CMD_REFRESH) {
 		/*Not supported*/
 		dbg("Not suported Proactive command");
 		return FALSE;
@@ -355,10 +355,11 @@ static TReturn s_envelope(CoreObject *o, UserRequest *ur)
 		return TCORE_RETURN_EINVAL;
 	}
 	for (count = 0; count < envelope_cmd_len; count++) {
-		dbg("envelope_cmd %02x", envelope_cmd[count]);
-		sprintf(pbuffer, "%02x", envelope_cmd[count]);
+		dbg("envelope_cmd %02hhX", envelope_cmd[count]);
+		sprintf(pbuffer, "%02hhX", envelope_cmd[count]);
 		pbuffer += 2;
 	}
+
 	dbg("pbuffer %s", envelope_cmdhex);
 	cmd_str = g_strdup_printf("AT+SATE=\"%s\"", envelope_cmdhex);
 	req = tcore_at_request_new(cmd_str, "+SATE:", TCORE_AT_SINGLELINE);
-- 
1.7.10.4

